analyzers/tests/SonarAnalyzer.UnitTest/TestCases/RedundancyInConstructorDestructorDeclaration.CSharp12.cs

```diff
@@ -0,0 +1,51 @@
+﻿// https://github.com/SonarSource/sonar-dotnet/issues/8092
+namespace Repro_8092
+{
+    namespace PrimaryParameterlessConstructor
+    {
+        class AClassWithBody() { }                // FN
+        class AClassWithoutBody();                // FN
+        struct AStructWithBody() { }              // FN
+        struct AStructWithoutBody();              // FN
+        record ARecordWithBody() { }              // FN
+        record ARecordWithoutBody();              // FN
+        record struct ARecordStructWithBody() { } // FN
+        record struct ARecordStructWithoutBody(); // FN
+
+        namespace FieldInitializerInStructRequiresConstructor
```

Comment:
Optional: should we leave a comment here to give some context?
`// https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/constructor-errors?f1url=%3FappId%3Droslyn%26k%3Dk(CS8983)#constructors-in-struct-types`